### ISSUE TITLE
Phase 2: MiniLM intent classifier — BERT-tiny zero-shot TFLite

### ISSUE THREAD

**Issue by lokhor** [NickMonrad/kernel-ai-assistant#359]
## Summary

Implement the `MiniLMIntentClassifier` as the Tier 2 semantic fallback when regex doesn't match. Plugs into `QuickIntentRouter` via the existing `IntentClassifier` interface.

## Model selection (research complete — see `docs/research/bert-intent-classifier-research.md`)

**all-MiniLM-L6-v2 int8 TFLite** (~15 MB bundled asset)
- ~10–15ms warm inference on Snapdragon 8 Gen 2 via XNNPACK (3× under 50ms budget)
- MTEB ~56, STS ~78 — best quality in class for zero-shot sentence similarity
- Handles informal/NZ/AU English well (trained on 1B+ diverse sentence pairs)
- TinyBERT and MobileBERT rejected — not sentence-embedding models, require fine-tuning

### New file: `core/skills/src/main/java/com/kernel/ai/core/skills/MiniLMIntentClassifier.kt`
- Implements `QuickIntentRouter.IntentClassifier`
- Hilt `@Singleton @Inject constructor(@ApplicationContext context: Context)`
- Loads `minilm-l6-v2-int8.tflite` + `vocab.txt` from assets
- BERT WordPiece tokenizer (pure Kotlin, no external lib)
- Mean-pool token embeddings + L2-normalise → 384-dim vector
- Pre-computes intent centroids (avg of 8 phrases) at startup on `Dispatchers.Default`
- Confidence threshold: **0.75** cosine similarity
- Ambiguity margin: **0.05** (gap between #1 and #2 intent must exceed this)
- Degrades gracefully — if model asset missing, `classify()` returns null (falls to E4B)

### New asset: `app/src/main/assets/intent_phrases.json`
All 20 intents with 8 NZ/AU English example phrases each (see research doc for schema). Phrases drive the centroid computation — adding a new intent = adding phrases, no retraining.

### New asset: `app/src/main/assets/minilm-l6-v2-int8.tflite` (~15 MB)
Download from Hugging Face or convert via ONNX pipeline (see research doc Section 3.1).
Alternatively: use pre-converted LiteRT model from `Bombek1/paraphrase-MiniLM-L6-v2-litert`.

### New asset: `app/src/main/assets/vocab.txt` (~230 KB)
Standard BERT 30,522-token WordPiece vocabulary from `bert-base-uncased`.

### Hilt wiring
Add binding in `core/skills/src/main/java/com/kernel/ai/core/skills/di/SkillsModule.kt` (create if needed):
```kotlin
@Binds @Singleton
abstract fun bindIntentClassifier(impl: MiniLMIntentClassifier): QuickIntentRouter.IntentClassifier
```

### Cold start strategy
Pre-compute embeddings on `Dispatchers.Default` at app startup (~800ms total). Any `classify()` calls before init completes return null (fall through to E4B — acceptable).

## Acceptance criteria
- [ ] `MiniLMIntentClassifier` compiles and implements `IntentClassifier` interface
- [ ] Model, vocab, and intent phrases bundled in assets
- [ ] Classifier correctly handles 20 intents from Phase 1B
- [ ] NZ/AU colloquial phrases (e.g. "torch on bro", "flick a text") route correctly
- [ ] Cold start happens on background thread — no ANR
- [ ] Falls through gracefully if assets missing
- [ ] Existing 130+ QuickIntentRouterTest tests still pass

## Related
- Parent: #353
- Research doc: `docs/research/bert-intent-classifier-research.md`
- Blocked by: Phase 1B (#357, merged)
- Blocks: #358 (Phase 3 integration)
